src/Cello/simulation_Simulation.cpp

```diff
@@ -985,9 +1013,30 @@ void Simulation::r_monitor_performance(CkReductionMsg * msg)
     }
   }
 
-  ASSERT2("Simulation::monitor_performance()",
-	  "Actual array length %d != expected array length %d",
-	  m,n, (m == n) );
+  const long long max_proc_blocks    = counters_reduce[m++]; // 10
+  const long long max_proc_particles = counters_reduce[m++]; // 11
+  
+  monitor()->print
+    ("Performance","simulation max-proc-blocks %d",  max_proc_blocks);
+  monitor()->print
+    ("Performance","simulation max-proc-particles %d", max_proc_particles);
+
+  const double avg_proc_blocks = 1.0*num_blocks_total/CkNumPes();
```

Comment:
As mentioned above, you're mixing precision on the right hand side.  Is that going to work out OK for all compilers?

Reply:
Answering my own question:  it seems that when you mix float/double and int, it always gets promoted to float/double.  So, the 1.0 ensures that the compiler will Do The Right Thing on the RHS.